### PR TITLE
HTTPServerRequest: Build urlURL using URLComponents

### PR DESCRIPTION
**Pull request by nethraravindran**
This increases the performance by ~2% while accessing`urlURL`

### CODE REVIEW

**Comment by pushkarnk** [comment]
@nethraravindran Bridging UInt16 to NSNumber doesn't work pre-4.2. We have two options:
1. replace it by `NSNumber(value:)` all throughout
2. do (1) only for pre-4.2 compiler versions

I ran a small benchmark and find performance of `NSNumber(value:)` similar to `x as NSNumber`. So, we may adopt (1).

File: Sources/KituraNet/HTTP/HTTPServerRequest.swift [hunk 1]
```diff
@@ -41,18 +41,24 @@ public class HTTPServerRequest: ServerRequest {
 
     @available(*, deprecated, message: "URLComponents has a memory leak on linux as of swift 3.0.1. use 'urlURL' instead")
     public var urlComponents: URLComponents {
-        return URLComponents(url: urlURL, resolvingAgainstBaseURL: false) ?? URLComponents()
+        if let _urlComponents = _urlComponents {
+            return _urlComponents
+        } else {
+            return URLComponents(url: urlURL, resolvingAgainstBaseURL: false) ?? URLComponents()
```
Comment (pushkarnk) [inline Sources/KituraNet/HTTP/HTTPServerRequest.swift]: @nethraravindran We must cache it here as well, in the scenario where `urlComponents` was invoked before `urlURL`

File: Sources/KituraNet/HTTP/HTTPServerRequest.swift [hunk 2]
```diff
@@ -41,18 +41,24 @@ public class HTTPServerRequest: ServerRequest {
...
+    private var _urlComponents: URLComponents?
+
     public var urlURL: URL {
         if let _url = _url {
             return _url
         }
-        var url = ""
 
-        self.enableSSL ? url.append("https://") : url.append("http://")
+        _urlComponents = URLComponents()
```
Comment (pushkarnk) [inline Sources/KituraNet/HTTP/HTTPServerRequest.swift]: check if `_urlComponents` already exists here, we could have initialised it in `urlComponents`

File: Sources/KituraNet/HTTP/HTTPServerRequest.swift (hunk 1 above)
Comment (nethraravindran) [reply Sources/KituraNet/HTTP/HTTPServerRequest.swift]: @pushkarnk Sure!

**Pull request merged**